fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayContains.java

```diff
@@ -71,6 +71,15 @@ public <R, C> R accept(ExpressionVisitor<R, C> visitor, C context) {
 
     @Override
     public List<FunctionSignature> getSignatures() {
+        if (getArgument(0).getDataType().isArrayType()
+                &&
+                ((ArrayType) getArgument(0).getDataType()).getItemType()
+                        .isSameTypeForComplexTypeParam(getArgument(1).getDataType())) {
+            // return least common type
+            return ImmutableList.of(
+                    FunctionSignature.ret(BooleanType.INSTANCE)
+                            .args(ArrayType.of(new AnyDataType(0)), new AnyDataType(0)));
```

Comment:
could use another static field to avoid generate new object each time

Reply:
done!